pkg/asset/ignition/node.go

```diff
@@ -1,44 +1,71 @@
 package ignition
 
 import (
+	"fmt"
 	"path/filepath"
 
-	ignition "github.com/coreos/ignition/config/v2_2/types"
+	"github.com/clarketm/json"
+	"github.com/coreos/ignition/v2/config/util"
+	igntypes "github.com/coreos/ignition/v2/config/v3_1/types"
 	"github.com/vincent-petithory/dataurl"
+	"k8s.io/apimachinery/pkg/runtime"
 
 	"github.com/openshift/installer/pkg/asset"
 )
 
 // FilesFromAsset creates ignition files for each of the files in the specified
 // asset.
-func FilesFromAsset(pathPrefix string, username string, mode int, asset asset.WritableAsset) []ignition.File {
-	var files []ignition.File
+func FilesFromAsset(pathPrefix string, username string, mode int, asset asset.WritableAsset) []igntypes.File {
+	var files []igntypes.File
 	for _, f := range asset.Files() {
 		files = append(files, FileFromBytes(filepath.Join(pathPrefix, f.Filename), username, mode, f.Data))
 	}
 	return files
 }
 
 // FileFromString creates an ignition-config file with the given contents.
-func FileFromString(path string, username string, mode int, contents string) ignition.File {
+func FileFromString(path string, username string, mode int, contents string) igntypes.File {
 	return FileFromBytes(path, username, mode, []byte(contents))
 }
 
 // FileFromBytes creates an ignition-config file with the given contents.
-func FileFromBytes(path string, username string, mode int, contents []byte) ignition.File {
-	return ignition.File{
-		Node: ignition.Node{
-			Filesystem: "root",
-			Path:       path,
-			User: &ignition.NodeUser{
-				Name: username,
+func FileFromBytes(path string, username string, mode int, contents []byte) igntypes.File {
+	return igntypes.File{
+		Node: igntypes.Node{
+			Path: path,
+			User: igntypes.NodeUser{
+				Name: &username,
 			},
+			Overwrite: util.BoolToPtr(true),
```

Comment:
what was the behavior with spec2 vs spec3 with overwriting? would love to see that in the commit message.

Reply:
+1.
Spec2 defaults to Overwrite: true, spec3 to false, so we need to set it here explicitly